libFMS.F90

```diff
@@ -172,14 +172,18 @@ module fms
                          fm_util_reset_caller, fm_util_set_no_overwrite, &
                          fm_util_reset_no_overwrite, fm_util_set_good_name_list, &
                          fm_util_reset_good_name_list, fm_util_get_length, &
-                         fm_util_get_integer, fm_util_get_logical, fm_util_get_real, &
+                         fm_util_get_integer, fm_util_get_logical, &
+                         fm_util_get_real, &
                          fm_util_get_string, fm_util_get_integer_array, &
-                         fm_util_get_logical_array, fm_util_get_real_array, &
+                         fm_util_get_logical_array, &
+                         fm_util_get_real_array, &
                          fm_util_get_string_array, fm_util_set_value, &
                          fm_util_set_value_integer_array, fm_util_set_value_logical_array, &
-                         fm_util_set_value_real_array, fm_util_set_value_string_array, &
+                         fm_util_set_value_real_array_r4, fm_util_set_value_real_array_r8, &
+                         fm_util_set_value_string_array, &
                          fm_util_set_value_integer, fm_util_set_value_logical, &
-                         fm_util_set_value_real, fm_util_set_value_string, &
+                         fm_util_set_value_real_r4, fm_util_set_value_real_r8, &
+                         fm_util_set_value_string, &
```

Comment:
i think it would be better to leave this as is, it would be better to add interfaces instead of forcing people to use the r4/r8 ones here.

Reply:
Sneaky, it's actually used, it's used for error checking in reads

Reply:
Yeah they don't need to be part of the bigger interface, the set_value_real r4/r8 versions can be in their own interface called `fm_util_set_value_real` so there isn't a change to the name. This way we won't have to change the calls in the components and coupler.

Reply:
I dont know why the "Sneaky" comment is here, that was for the "99 continue" comment.  But resolved, removed the _r4 _r8 versions

Reply:
In order to use just `fm_util_set_value_real`, a separate interface `fm_util_set_value_real` using the *_r4 and *_r8 versions of the procedures is required.  But `fm_util_set_value` interface needs to use the `fm_util_set_value_real`.  It seems like fortran can't do interface within interfaces and we're stuck with this for now..

Reply:
That said, I don't know why the "sneaky" comment is here.  That was for the "99 continue" comment

Reply:
Sorry my reply before wasn't really accurate, they should still be in the interface but you should be able to have both the bigger generic one and the added r4/r8 one. 

You don't need to add the fm_util_set_value_real interface to the existing one, you can just add the same module procedure statements to both interfaces and it'll work the same way.

Reply:
got it, wow it took me long.  fixed, created new interfaces `fm_util_set_value_real` and `fm_util_set_value_real_array` and removed _r4 and _r8 versions from libFMS.F90